WirescaleCli/WireguardConfigurationWriter.cs

```diff
@@ -0,0 +1,47 @@
+﻿using System.Diagnostics;
+using System.Text;
+
+namespace WirescaleCli;
+
+public class WireguardConfigurationWriter
+{
+    private readonly string _configurationFilePath;
+
+    public WireguardConfigurationWriter(string configurationFilePath)
+    {
+        _configurationFilePath = configurationFilePath;
+    }
+
+    public void Write(WireguardKeyPair wireguardKeyPair, WirescalePeerRegistration peerRegistration)
+    {
+        StringBuilder configurationBuilder = new StringBuilder();
+
+        configurationBuilder.AppendLine("[Interface]");
+        configurationBuilder.AppendLine($"PrivateKey = {wireguardKeyPair.PrivateKey}");
+        configurationBuilder.AppendLine($"Address = {peerRegistration.AllowedIps.Single()}");
+
+        configurationBuilder.AppendLine();
+
+        configurationBuilder.AppendLine("[Peer]");
+        configurationBuilder.AppendLine($"PublicKey = {peerRegistration.ServerPublicKey}");
+        configurationBuilder.AppendLine($"AllowedIPs = {peerRegistration.ServerAllowedIps.Single()}");
+        configurationBuilder.AppendLine($"Endpoint = {peerRegistration.ServerEndpoint}");
+
+        File.WriteAllText(_configurationFilePath, configurationBuilder.ToString());
+
+        //Move wireguard logic/configuration elsewhere
+        var process = Process.Start(new ProcessStartInfo
+        {
+            FileName = "wg-quick",
+            Arguments = $"up ./{_configurationFilePath}",
+            UseShellExecute = false,
+            CreateNoWindow = true,
+        });
+
+        process.WaitForExit();
+        if (process.ExitCode != 0)
+        {
+            throw new Exception("Failed to configure WireGuard.");
```

Comment:
Similar feedback to the above, how do I know what failed and how to fix it?

Reply:
It's relatively safe to assume a non-zero exit code here is caused by the interface already being registered, so for now I've added a more specific exception type (handled and written to stderr in `Program.cs`,)[which provide a couple of possible solutions](https://github.com/runesoerensen/wirescale/pull/5/commits/1cbc110a2ac31ed8f248908b40a9eb6d67abec30).